pkg/factory/risingwave_object_factory.go

```diff
@@ -1088,14 +1088,14 @@ func (f *RisingWaveObjectFactory) envsForS3() []corev1.EnvVar {
 			endpoint = "https://" + endpoint
 		}
 
-		return envsForS3Compatible(s3Spec.Region, endpoint, s3Spec.Bucket, s3Spec.RisingWaveS3Credentials)
+		return envsForS3Compatible(s3Spec.Region, endpoint, s3Spec.Bucket, s3Spec.RisingWaveS3Credentials, s3Spec.ForcePathStyle)
 	}
 
 	// AWS S3 mode.
 	return envsForAWSS3(s3Spec.Region, s3Spec.Bucket, s3Spec.RisingWaveS3Credentials)
```

Comment:
It would be better to keep the behaviour aligned between S3 and S3 compatible backends. Let's add the env for S3 as well.